### PR TITLE
new progressbar component of claim branch && more values available

### LINKED ISSUE #2008: Progress bar with percentages
**Description**

Create a progress bar component with percentages on top.
Clicking on the percentages should update the bar.

Screenshot from 2021-12-17 13-15-24

**Proposal**

This component will later be used in the claiming page.
The value will be read from and persisted to input state (to be defined).
Bar should be updated on percentage click as well as manual input change.

Maybe this state will be stored on redux? Still not defined.

For further context, this what what the surrounding page will look like:
Screenshot from 2021-12-17 13-34-05

**Additional context**
- Wireframes: https://www.figma.com/file/DPp3awAUcqpwpV1dD4wFB9/CoW-claiming?node-id=92%3A5718

### PR DESCRIPTION
**Pull request by maria-vslvn**
# Summary

Fixes #2008 

New Progress Bar component was created, values can be set by clicking the numbers at the top of the bar or via dragging the bar or clicking at any bar point.
image

  # To Test

1. Open the page `Profile`
The bar is at the bottom of the form

### CODE REVIEW

File: src/custom/components/ProgressBar/styled.tsx [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+export const ProgressBarWrap = styled(FlexWrapMod)`
+  max-width: 500px; //optional
+  padding-top: 40px;
+  position: relative;
+`
+
+export const ProgressContainer = styled.div`
+  background-color: ${({ theme }) => transparentize(0.61, theme.text1)};
+  height: 16px;
```
Comment (alongoni) [inline src/custom/components/ProgressBar/styled.tsx:19]: Hey @maria-vslvn, I love it! Works fine! 
Maybe we can add 24px height in order to have more space inside the bar and increase the font size of the variable %. (to `font-size: 16px;` and `font-weight: 900;`)

Another thing, we can decrease a bit the font-size of the values at the top (to 12px).

It'll look like this:
image

**Review by alfetopito — COMMENTED** [review]
Excellent work Maria, looks perfect!

While it does make sense for the component in isolation that's not how we envision the component to be used regarding the state handling.

Could you please remove the internal state and add a callback as a parameter like in the proposed API on the issue https://github.com/gnosis/cowswap/issues/2008#issuecomment-1006848257 named `onPercentageClick` or `onClick`.

Then, instead of using the `setProgressVal(value)` you'd call `onPercentageClick(value)`.

The parent component is the who'll hold the state and update the ProgressBar component by passing the updated `value` component.

You can have a test parent component, but it's not required for the current ProgressBar component.

Let me know if you have any questions

**Comment by maria-vslvn** [comment]
@alfetopito please check the updates

File: src/custom/components/ProgressBar/index.tsx [hunk 2]
```diff
@@ -0,0 +1,62 @@
+import { ProgressBarWrap, ProgressContainer, Progress, Label, FlexWrap, HiddenRange, ProgressVal } from './styled'
+
+interface progressBarProps {
```
Comment (alfetopito) [inline src/custom/components/ProgressBar/index.tsx:3]: Nitpick: can you make the interface name capitalized?

That's a (non written?) standard we use in the project

```suggestion
interface ProgressBarProps {
```

File: src/custom/pages/Profile/index.tsx [hunk 3]
```diff
@@ -206,6 +209,7 @@ export default function Profile() {
           </GridWrap>
           {!account && <Web3Status openOrdersPanel={() => console.log('TODO')} />}
         </GridWrap>
+        <ProgressBar onPercentageClick={() => setPercentage(111)} percentage={percentage} />
```
Comment (alfetopito) [inline src/custom/pages/Profile/index.tsx:212]: Not quite there yet.

It should be like this:

```suggestion
        <ProgressBar onPercentageClick={setPercentage} percentage={percentage} />
```

Then now the state management is on this component, but triggered in the child via the callback

**Review by alfetopito — COMMENTED** [review]
You are very close!

**Comment by elena-zh** [comment]
Scrollbar LGTM itself, but is this OK that it becomes 100% filled when I select any option, including 0%?
And it is impossible to change a value then
image

**Comment by anxolin** [comment]
It would be great to have this PR merged soonish, we are arriving the point where we need it

**Comment by ramirotw** [comment]
> It would be great to have this PR merged soonish, we are arriving the point where we need it

please take a look in the Profile page

> Scrollbar LGTM itself, but is this OK that it becomes 100% filled when I select any option, including 0%?
> And it is impossible to change a value then

the issue should be fixed

**Review by ramirotw — APPROVED** [review]

**Review by alongoni — APPROVED** [review]
Great job!

**Comment by elena-zh** [comment]
Changes LGTM!

**Pull request merged**